### DISCUSSION TITLE
lucid: What a best way to serialize withCount

### DISCUSSION THREAD

**Discussion by yariksav** [adonisjs/core#1613] [Help]
What a best wat to serialize withCount

Now after
```
return Product.query().withCount('items').paginate(page)
```
lucid put items_count into $extras and doesn't serialize when prepare response

For this I need to parse products manually or make computed param in Product model

```
  @computed()
  public get itemsCount () {
    return this.$extras.items_count
  }
```

@adonisjs/lucid@8.4.2
Thanks!

**Comment by thetutlage** [comment]
Yes. This is how it is expected to be done. Lucid doesn't serialize everything, since it is very much possible that you may have sensitive data inside the `$extras` property or some property inside the `$extras` object has the same name as the model own property.

Also, creating a computed property makes it more explicit and easy to reason about. Since anyone looking at the model can see that `itemsCount` can appear in some responses.

**Reply by yariksav** [reply]
Thank you very much!
